lib/auth/oidc.go

```diff
@@ -103,25 +103,7 @@ func (a *Server) createOIDCClient(conn services.OIDCConnector) (*oidc.Client, er
 				"unknown problem with connector %v, most likely URL %q is not valid or not accessible, check configuration and try to re-create the connector",
 				conn.GetName(), conn.GetIssuerURL())
 		}
-		if err := a.emitter.EmitAuditEvent(ctx, &events.UserLogin{
-			Metadata: events.Metadata{
-				Type: events.UserLoginEvent,
-				Code: events.UserSSOLoginFailureCode,
-			},
-			Method: events.LoginMethodOIDC,
-			Status: events.Status{
-				Success:     false,
-				Error:       trace.Unwrap(ctx.Err()).Error(),
-				UserMessage: err.Error(),
-			},
-		}); err != nil {
-			log.WithError(err).Warn("Failed to emit OIDC login failure event.")
-		}
-		// return user-friendly error hiding the actual error in the event
-		// logs for security purposes
-		return nil, trace.ConnectionProblem(nil,
-			"failed to login with %v",
-			conn.GetDisplay())
```

Comment:
Since we no longer show the users the actual login errors (but a generic login failed msg), I didn't think it was necessary to hide this error and simply return the actual error for the outer caller to emit the events (also removes double emitting)